src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,355 @@
+        }
+        throw new CloudException(String.format("The cluster is not deleted in %d millisecond", timeoutInMillisecond));
+    }
+
+    private String getBearerToken(String apiKey, String apiSecret) throws CloudException {
+        try {
+            String query = String.format("{ \"query\": \"mutation { login(apiKey: \\\"%s\\\", apiSecret: \\\"%s\\\") {token} }\" }", apiKey, apiSecret);
+            RequestBody body = RequestBody.create(JSON, query);
+            Request request = new Request.Builder()
+                    .url(HttpUrl.get(uri))
+                    .post(body)
+                    .header("Content-Type", "application/json")
+                    .build();
+            call = client.newCall(request);
+            Response response = call.execute();
+            String loginResponse = response.body().string();
+            LOG.info(maskValueOfToken(loginResponse));
+            return mapper.readTree(loginResponse).get("data").get("login").get("token").asText();
+        } catch (Exception e) {
+            throw new CloudException("Get bearer token is failed");
+        }
+    }
+
+    // It is not possible to connect a cluster with its actual name, it seems a bug of the cloud API, that method is a workaround.
+    // There could be a better way to get this name, TODO: contact with cloud team.
+    private String getConnectionName(String clusterId) throws CloudException {
+        if (baseUrl.contains("uat.hazelcast.cloud"))
+            return String.format("ua-%s", clusterId);
+        else if (baseUrl.contains("cloud.hazelcast.com"))
+            return String.format("pr-%s", clusterId);
+        else {
+            throw new CloudException("Given base url isn't known and connection name cannot be created");
+        }
+    }
+
+    // It replaces the  value of token with *** in a response body. Such as bearer token, discovery token.
+    private String maskValueOfToken(String json)
+    {
+        return json.replaceAll("\"token\":\"(.*?)\"", "\"token\":\"******\"");
+    }
+
+    // It creates a folder with name clusterId under /home/user/
+    // Then it downloads the certificates for the cluster and put them in the created folder
+    private String downloadCertificatesAndGetPath(String clusterId) throws CloudException {
+        try
+        {
+            String userHome = System.getProperty("user.home");
+            Path pathClusterId = Paths.get(userHome, clusterId);
+
+            // If folder with clusterId is there then certificates are already downloaded
+            if(!Files.exists(pathClusterId))
+            {
+                createFolderAndDownloadCertificates(pathClusterId, clusterId);
+            }
+            return Paths.get(pathClusterId.toString(), "certificates").toString() + File.separator;
+        }
+        catch(Exception e)
+        {
+            throw new CloudException("Problem occurred during certificates download, error message is: " + e.getMessage());
```

Comment:
Why don't you reuse HazelcastException?
```suggestion
            throw new HazelcastException("Problem occurred during certificates download, error message is: ", e);
```

Reply:
But we are downloading certificates from Cloud API, shouldn't it be a CloudException?
I used CloudException as a general exception type of Cloud API.